server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -49,179 +54,271 @@
+    @Value("${motif.search.buffer.size:16000000}")
+    private int bufferSize;
+
+    @Value("${motif.search.sliding.window:1000}")
+    private int defaultOverlap;
+
+    @Value("${motif.search.page.size:100}")
+    private int defaultPageSize;
+
+    @Value("${motif.search.include.sequence:false}")
+    private boolean defaultIncludeSequence;
+
+    @Value("${motif.search.result.size.limit:2000000}")
+    private int searchResultSizeLimit;
 
     @Autowired
     private ReferenceGenomeManager referenceGenomeManager;
 
+    @Autowired
+    private ReferenceManager referenceManager;
+
+
     public Track<StrandedSequence> fillTrackWithMotifSearch(final Track<StrandedSequence> track,
                                                             final String motif,
                                                             final StrandSerializable strand) {
         final List<StrandedSequence> result = search(
                 MotifSearchRequest.builder()
+                        .referenceId(track.getId())
                         .startPosition(track.getStartIndex())
                         .endPosition(track.getEndIndex())
                         .chromosomeId(track.getChromosome().getId())
                         .motif(motif)
                         .searchType(MotifSearchType.REGION)
-                        .pageSize(0)
+                        .pageSize(Integer.MAX_VALUE)
                         .strand(strand)
                         .build()).getResult().stream()
-                .map(m -> new StrandedSequence(m.getStart(), m.getEnd(), m.getValue(), m.getStrand()))
+                .map(m -> new StrandedSequence(m.getStart(), m.getEnd(), m.getSequence(), m.getStrand()))
                 .collect(Collectors.toList());
         track.setBlocks(result);
         return track;
     }
 
     public MotifSearchResult search(final MotifSearchRequest request) {
         verifyMotifSearchRequest(request);
+        Reference reference = referenceGenomeManager.getOnlyReference(request.getReferenceId());
+        if (reference == null) {
+            throw new IllegalStateException(getMessage(
+                    MessagesConstants.ERROR_REFERENCE_READING, request.getReferenceId()));
+        }
+        reference.setChromosomes(referenceGenomeManager.loadChromosomes(request.getReferenceId()));
         switch (request.getSearchType()) {
             case WHOLE_GENOME:
-                return searchWholeGenomeMotifs(request);
+                return searchWholeGenomeMotifs(request, reference);
             case CHROMOSOME:
-                return searchChromosomeMotifs(request);
+                return searchChromosomeMotifs(request, reference);
             case REGION:
-                return searchRegionMotifs(request);
+                return searchRegionMotifs(request, reference);
             default:
                 throw new IllegalStateException("Unexpected search type: " + request.getSearchType());
```

Comment:
Maybe it makes sense to form a message using getMessage().

Reply:
`+`